docs/config_entries_config_flow_handler.md

```diff
@@ -158,3 +158,27 @@ Translations for the config flow handlers are defined under the `config` key in
 ```
 
 When the translations are merged into Home Assistant, they will be automatically uploaded to [Lokalise](https://lokalise.co/) where the translation team will help to translate them in other languages. While developing locally, you will need to run `python3 -m script.translations develop` to see changes made to `strings.json` [More info on translating Home Assistant.](translations.md)
+
+## Migration to a new version of a Config Entry
+
+As mentioned above - each Config Entry has a version assigned to it. This is to be able to migrate Config Entry data to new formats when Config Entry schema changes. 
+Migration can be handled programatically by implementing function `async_migrate_entry` in your component's `__init__.py` file. The function should return `True` if migration is successfull.
+Example code:
+```python
+async def async_migrate_entry(hass, config_entry: ConfigEntry):
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version == 1:
+
+        new = {**config_entry.data}
+        # TODO: modify Config Entry data
+        
+        config_entry.data = {**new}
```

Comment:
The config entry data should be update via `hass.config_entries.async_update_entry`.
```suggestion
        hass.config_entries.async_update_entry(config_entry, data=new)
```

Reply:
Assigning directly to data is correct during a migration.

Reply:
To avoid a save? We're changing the attribute type like this.

Reply:
A config entry is automatically saved when a migration is done.

Reply:
Yes. Are we trying to avoid an extra save by not using our standard interface method?

Reply:
The external save method has other side effects that we don't want right now, like notifying listeners.

Reply:
Ok. Maybe we need an interface on the `ConfigEntries` class for migration?

It's weird that we don't follow our standard to use a config entries interface for the migration while for everything else regarding config entry manipulation we do that.